hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -624,8 +629,23 @@ def generate_new_examples(self):
 
         self.health_check_state = HealthCheckState()
 
+        def should_generate_more():
+            if not self.interesting_examples:
+                return True
+            # We've found some bugs, but it's not much time since the first was found
+            # and not many examples since the most last new bug was found (or at all).
+            return (
+                benchmark_time() < self.stop_time
+                # We use call_count rather than valid_examples to avoid looping for
+                # too long if there are very few valid examples, e.g. from st.just()
+                and self.call_count < max(10, self.last_bug_calls * 2)
```

Comment:
Would it make more sense to combine these two `call_count` checks into a single one with `min`?

Reply:
If you mean "should this `max` be `min`", nope - the idea is that if we find a bug on the very first call, we still try ten before stopping.  If you mean "should this line be combined with the other", the other is (now) using `first_bug_calls` so it's incommensurate.